### ISSUE TITLE
default currency format is unusable with multiple specific iOS mobile keyboards

### ISSUE THREAD

**Issue by NeOMakinG** [shapeshift/web#7060]
### Overview

On my french phone, with the default currency number formatter, I can't use any swap/bridge/send because I don't have access to the `dot` key on the keyboard which is required for amounts under 0 (eg 0.1 ETH, 0.0001 BTC):
<img width="272" alt="image" src="https://github.com/shapeshift/web/assets/14963751/01ae07c3-dbe9-4050-8d98-ef52ee6567b6">

If I select another format:
<img width="403" alt="image" src="https://github.com/shapeshift/web/assets/14963751/f7debc7c-8309-4c05-91c7-5aa75f24b1c3">

It's not working fine as it uses coma for amounts under 0

### References and additional details

We might want to adapt the default currency format based on the local lang or change the keyboard is it's technically possible.

### Acceptance Criteria

I'm able to perform swap/bridge/send operations by default right after installing the mobile application if I have a french or any other keyboards using commas as under 0 separator

**Comment by NeOMakinG** [comment]
@firebomb1 you might be interested by this one as well

**Comment by gomesalexandre** [comment]
I'm able to repro as well:

- English *locale* in phone
- GBoard with EN/FR (in this order)
- EU *region*

And despite the two former, still seing the comma formatter, so assuming region takes precedence here?

**Comment by NeOMakinG** [comment]
I couldn't reproduce on android as the android keyboard has both comma and dots

**Comment by firebomb1** [comment]
Thanks for the ping! Same, no issue on Android. I'm chatting with @NeOMakinG on Discord as I don't have an iOS or macOS device to test this, but I've suggested few solutions we can test.

**Comment by NeOMakinG** [comment]
After investigation:

Keyboard display depends on the Region: 

US region uses the `.` separator

<img width="333" alt="image" src="https://github.com/shapeshift/web/assets/14963751/708cee97-ea16-4fa5-a302-186d3fe1331c">

while if as an example you are french, you have the `,` separator

There are no ways to tweak that for now, except using the `number` or `text` input type with a decimal allowance pattern, which would allow some other characters than numbers, ',' and '.'

**Comment by 0xean** [comment]
https://github.com/zoontek/react-native-localize - maybe able to leverage something like this

**Comment by NeOMakinG** [comment]
Finally after further investigations, and thanks to @0xean suggestion, we have multiple choices, one particular looks very interesting to me:

- `react-native-localize` permits to get the number formatting selected in the settings (https://github.com/zoontek/react-native-localize?tab=readme-ov-file#getnumberformatsettings), we could leverage this to change the default settings of our formatter based on the number formatting selected on the device, be careful we might need to handle the fact that the user changed the app settings and so display another keyboard in that case?
- Just disable the numeric keyboard on iOS

More context:

Changing the "Region" change the number format by default and you can change the number format if you want
https://github.com/zoontek/react-native-localize?tab=readme-ov-file#getnumberformatsettings

**Comment by 0xean** [comment]
> react-native-localize permits to get the number formatting selected in the settings (https://github.com/zoontek/react-native-localize?tab=readme-ov-file#getnumberformatsettings), we could leverage this to change the default settings of our formatter based on the number formatting selected on the device, be careful we might need to handle the fact that the user changed the app settings and so display another keyboard in that case? (We could simply add a formatter field to the window object and so the shapeshift app would detect it to change the default formatter)

Can you give an idea of the level of effort to get this done?

**Comment by NeOMakinG** [comment]
Honestly it's pretty simple, I would say `1`:
Here are the 4 settings we need to push to the `window` object through the webview:
- `{ groupingSeparator: ",", decimalSeparator: "." }`
- `{ groupingSeparator: ".", decimalSeparator: "," }`
- `{ groupingSeparator: " ", decimalSeparator: "." }`
- `{ groupingSeparator: " ", decimalSeparator: "," }`

if one of these 4 settings are existing on `window` on the shapeshift web side, change the default `CurrencyFormats` to the matching one (we will need a mapping between thoses 4 settings and 4 default CurrencyFormats, a simple object to map them)

![image](https://github.com/shapeshift/web/assets/14963751/a49dfe7f-16e9-4d4f-9ee0-0c994cb43d7c)

We need a way to use the `decimal` input mode if the current setting matches the default setting with the `window` attribute, or use the base keyboard with more than numbers if the window attribute exists and the default settings doesn't match. For this, I would probably introduce a `useNumberInputMode` hook to consume inside all those occurrences

TLDR: 2 small PR, one on mobile side, one on web side, the biggest one is on web side: change the default setting matching the window attribute, then when you render any number input you have to choose between "the decimal inputMode is usable or nop it's too risky let's switch to the basic keyboard"

**Comment by 0xean** [comment]
Okay, if its gonna take a mobile PR, lets get it done because releasing is a pain and we can roll it into the current release. This essentially is a bug, so doesn't need to go through the typical grooming process.

**Comment by NeOMakinG** [comment]
> Okay, if its gonna take a mobile PR, lets get it done because releasing is a pain and we can roll it into the current release. This essentially is a bug, so doesn't need to go through the typical grooming process.

Ok cool, jumping on it right now

**Comment by NeOMakinG** [comment]
Finally, thanks to @gomesalexandre we found that our library accepts `allowedDecimalSeparators`... It's way simpler than expected then
